### ISSUE TITLE
Missing focus indicators

### ISSUE THREAD

**Issue by hostle83** [DSpace/dspace-angular#3084]
**Describe the bug**
When navigating the sandbox or demo site by keyboard-only, two instances are showing a missing focus indicator: 
1. The search icon in the utility navigation:
![image](https://github.com/DSpace/dspace-angular/assets/1933584/26decf9f-db79-4e10-98d0-3ef2dcd5b067)
 
2. The hierarchical tree view:
![image](https://github.com/DSpace/dspace-angular/assets/1933584/fe7e3486-9554-4dba-b0ce-e1a6fd62a1cc)

**To Reproduce**
1. Navigate the demo or sandbox site by keyboard only. 

**Expected behavior**
According to SC 2.4.7 Focus Visible (Level AA) (https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html), a vertical bar should be displayed in the field, indicating that the user can insert text. This happens, when the user clicks (or hits enter) on the search icon, but not before

In the hierarchical tree view, there is are no focus elements around the subject categories when navigating with keyboard only.

**Comment by alexandrevryghem** [comment]
@hostle83: I was able to reproduce the error in the search facets, currently there is a deadlock preventing to access that link so I added that fix to [this PR](https://github.com/DSpace/dspace-angular/pull/3065). For the first issue I can see a focus indicator on the latest version of DSpace, so could you perhaps check if this still happens for you?
<img src="https://github.com/DSpace/dspace-angular/assets/43750381/12961d58-831f-483a-b4a3-69f85e2a3176" width="40px">
